### ISSUE TITLE
Potential bug in Tie-breaking algorithm

### ISSUE THREAD

**Issue by williamledoux** [MieuxVoter/majority-judgment-api-python#48]
Hey,

Let a poll with 2 options (A and B), 6 mentions and 19 voters, 
OptionA: 1 | 8 | 7 | 1 | 1 | 1
OptionB: 6 | 3 | 4 | 4 | 1 | 1

According to my understanding of each of the two tie-breaking algorithms explained on the [wikipedia page](https://fr.wikipedia.org/wiki/Jugement_majoritaire) (the original [by Balinski et Laraki](https://1984f707-a-62cb3a1a-s-sites.googlegroups.com/site/ridalaraki/xfiles/BalinskiLarakiOR2014.pdf?attachauth=ANoY7cqPWnMx13pqteVZKJR34BsOOOHxkYJ-jpkygXCDrwlwIXesxHHT5-i97wa2rDzVW7SHy8Ttl_ygox1CJ-9_W9n7D-D8NVkX0oN6Sec6bo-dSUOiZHp084KaFEo9SiZRvjicvlNN6YECMHiGoOCR8QlBt1YntPKo96l3zbfjuxYYYBbofxqOM6FMS8_PX_ZkVeB39bLllltqils_-hqIOuiPGCc7En7UrIjAM2Bj1nevamSEvcI%3D&attredirects=0)  and the one [by Adrien Fabre](https://raw.githubusercontent.com/bixiou/highest_median/master/Tie-breaking%20Highest%20Median%20-%20Fabre%202019.pdf)), option B should be preferred to option A, but the [online tool](https://app.mieuxvoter.fr/) prefers A to B.

Here is this poll in the app:
https://app.mieuxvoter.fr/result/SMKafosuqv6qRpEEbGJj

Is that a bug or am I not understanding the tie breaking algorithm ?
Thank you

**Comment by domi41** [comment]
> Is that a bug or am I not understanding the tie breaking algorithm ?

**TLDR; Not a bug.**

When both the _adhesion_ and _contestation_ groups have the same size, we usually **favor the contestation**.

![Image](https://github.com/user-attachments/assets/caedb74a-01cb-419c-bac0-620181cf2aab)

Here both groups of B have size 6.
We favor contestation, and so the **contestation group of B is decisive**, and B ends up second.

We had to choose whether to favor adhesion or contestation in case of a tie in the web app.
It is essentially a political choice, not a mathematical one.

You could choose the other way, so long as you and you voters agree on the choice in advance.

It does make some sense to favor the contestation though ; we are conservative creatures, and any adhesion is usually less strongly felt than contestation.

---

Sorry for the long delay in the response ⏲  !
